pkg/sql/information_schema.go

```diff
@@ -2470,30 +2472,54 @@ func forEachTableDescWithTableLookupInternalFromDescriptors(
 	return nil
 }
 
+type roleOptions map[string]string
+
+func (r roleOptions) noLogin() bool {
+	_, noLogin := r["NOLOGIN"]
+	return noLogin
+}
+
+func (r roleOptions) validUntil(p *planner) (*time.Time, error) {
+	rolValidUntilStr, ok := r["VALID UNTIL"]
+	var rolValidUntil *time.Time = nil
+	if ok {
+		validUntil, _, err := pgdate.ParseTimestamp(
+			p.EvalContext().GetRelativeParseTime(),
+			p.EvalContext().GetDateStyle(),
+			rolValidUntilStr,
+		)
+		rolValidUntil = &validUntil
+		if err != nil {
+			return nil, errors.Errorf("rolValidUntil string %s could not be parsed", rolValidUntilStr)
+		}
+	}
+	return rolValidUntil, nil
+}
+
+func (r roleOptions) createDB() bool {
+	_, createDB := r["CREATEDB"]
+	return createDB
+}
+
 func forEachRole(
 	ctx context.Context,
 	p *planner,
-	fn func(username security.SQLUsername, isRole bool, noLogin bool, rolValidUntil *time.Time) error,
+	fn func(username security.SQLUsername, isRole bool, options roleOptions, settings tree.Datum) error,
 ) error {
 	query := `
 SELECT
 	u.username,
 	"isRole",
-	EXISTS(
-		SELECT
-			option
-		FROM
-			system.role_options AS r
-		WHERE
-			r.username = u.username AND option = 'NOLOGIN'
-	)
-		AS nologin,
-	ro.value::TIMESTAMPTZ AS rolvaliduntil
+  drs.settings,
+	json_object_agg(COALESCE(ro.option, 'null'), ro.value)::TEXT
```

Comment:
can we return the JSON value here?  then use the `Exists` in` json.JSON` to do the comparison work, instead of deserializing to this map?

Reply:
Well, the code above are helpers as it may become tedious to do the same logic over the functions that needs these role options. What I can do is to have this as a JSON value as you suggest instead of a map, I just need to figure it out how to do that.